plugins/module_utils/entity.py

```diff
@@ -156,14 +161,30 @@ def list(
         url = self.base_url if use_base_url else self.base_url + "/list"
         if endpoint:
             url = url + "/{0}".format(endpoint)
-        return self._fetch_url(
-            url,
-            method="POST",
-            data=data,
-            raise_error=raise_error,
-            no_response=no_response,
-            timeout=timeout,
-        )
+        entities_list = []
+        while True:
+            resp = self._fetch_url(
+                url,
+                method="POST",
+                data=data,
+                raise_error=raise_error,
+                no_response=no_response,
+                timeout=timeout,
+            )
+            entities_list.extend(resp[self.entity_type])
+            entities_count = len(entities_list)
+            data["offset"] = entities_count
+            if entities_count != self.entities_limitation:
+                break
+        custom_filters = self.module.params.get("custom_filter")
+        if custom_filters:
+            entities_list = self.filter_entities(entities_list, custom_filters)
+            entities_count = len(entities_list)
+
+        resp["entities"] = entities_list
```

Comment:
this should be changed from resp["entities"] = entities_list to resp[self.entity_type] = entities_list .